spec/models/report_spec.rb

```diff
@@ -157,86 +157,92 @@
+    end
+
+    it "appends the `with_roda_identifier` scope" do
+      report.reportable_activities
+
+      expect(query).to have_received(:with_roda_identifier)
+    end
+
+    it "returns the active_relation" do
+      expect(report.reportable_activities).to eq(active_relation)
     end
   end
 
   describe "#activities_created" do
-    it "only returns activities created in the given reporting period" do
-      fund = create(:fund_activity)
-      programme = create(:programme_activity, parent: fund)
+    let(:report) { build(:report, financial_quarter: 1, financial_year: 2020) }
+    let(:active_relation) { double("active_relation") }
+    let(:query) { double("query", where: active_relation) }
 
-      travel_to(Date.parse("2020-04-26")) do
-        @report = create(:report, organisation: build(:delivery_partner_organisation), fund: fund)
-        @project_in_reporting_period = create(:project_activity, parent: programme, organisation: @report.organisation)
-        @third_party_project_in_reporting_period = create(:third_party_project_activity, parent: @project_in_reporting_period, organisation: @report.organisation)
-      end
+    before do
+      allow(report).to receive(:reportable_activities).and_return(query)
+      allow(query).to receive(:activities_created).and_return(active_relation)
+    end
 
-      _project_outside_reporting_period = create(:project_activity, parent: programme, organisation: @report.organisation)
-      _third_party_project_outside_reporting_period = create(:third_party_project_activity, parent: @project_in_reporting_period, organisation: @report.organisation)
+    it "filters the #reportable_activities using the financial period" do
+      report.activities_created
 
-      expect(@report.activities_created).to match_array([
-        @project_in_reporting_period,
-        @third_party_project_in_reporting_period,
-      ])
+      expect(query).to have_received(:where)
+        .with(created_at: (Date.parse("01-Apr-2020")..Date.parse("30-Jun-2020")))
+    end
+
+    it "returns the active_relation" do
+      expect(report.activities_created).to eq(active_relation)
     end
   end
 
-  describe "forecasts" do
-    let(:report) { create(:report, financial_quarter: 1, financial_year: 2021) }
-    let(:programme) { create(:programme_activity, parent: report.fund) }
-
-    let!(:activities) do
-      2.times.map {
-        create(
-          :project_activity,
-          organisation: report.organisation,
-          parent: programme
-        ).tap do |activity|
-          ForecastHistory.new(
-            activity,
-            report: report,
-            financial_quarter: 2,
-            financial_year: 2021,
-            user: create(:delivery_partner_user)
-          ).set_value(50_000)
-        end
-      }
+  describe "#forecasts_for_reportable_activities" do
+    let(:report) { build(:report) }
+    let(:reportable_activities) { build_stubbed_list(:project_activity, 5) }
```

Comment:
I've not familiar with these `build_stubbed*` methods in FactoryBot. I need to have a proper look into their use...

thks!